frontend/src/components/shared/Button/styled-components.ts

```diff
@@ -71,6 +74,43 @@ function getSizeStyle(size: Size, theme: Theme): CSSObject {
   }
 }
 
+function getThemeStyle(
+  activeTheme: string,
+  baseTheme: string,
+  theme: Theme
+): CSSObject {
+  switch (activeTheme) {
+    case "Light":
+      return {
+        "&:hover": {
+          backgroundColor: theme.colors.gray20,
+        },
+      }
+    case "Dark":
+      return {
+        "&:hover": {
+          backgroundColor: theme.colors.gray90,
+        },
+      }
+    case "Custom Theme":
+      return {
+        "&:hover": {
+          backgroundColor: transparentize(theme.colors.primary, 0.5),
+        },
+      }
```

Comment:
This is more of a design question than a question about the code itself, but do we want to switch to using a transparentized primary color for all custom themes? It would feel a bit weird to me if the color of the menu buttons changed after I made a custom theme that leaves most of the colors the same but only changes the `textColor` or `font` options.

Currently, the way that we pick some "auxiliary" colors (such as the colors in the `st.info`/`st.warning`/`st.error` components) for a custom theme is to pick the same colors that we use in light/dark mode based on the luminance of the background color (see [this comment](https://github.com/streamlit/streamlit/blob/02cbb2244d1617e8b417ab2a7bb817691b973efc/frontend/src/theme/utils.ts#L480) and the function it lives in for how this is done). I feel like doing something similar here may make sense.

Reply:
Thanks for explaining this! I appreciate it, as I'm totally new to the codebase. I brought this and the `transparentDarkenedBgMix60` options to a design meeting with the Product folks, and they preferred the latter, which I believe also addresses the point you mentioned here: 

> It would feel a bit weird to me if the color of the menu buttons changed after I made a custom theme that leaves most of the colors the same but only changes the textColor or font options.

As it's based on the `bgColor`, and not the `primary` color anymore. I added that on [722a443](https://github.com/streamlit/streamlit/pull/4496/commits/722a443e7b282f1e066e7b20b5587473c9ebbb80), but let me know if you have any additional input!